docs/proposals/extracted_credentials_saved_as_secrets.md

```diff
@@ -0,0 +1,57 @@
+## Extracted Credentials Saved As Secrets
+
+Extracted Credentials are currently saved in our etcd for the service broker. This is not desirable for many reasons, but the two biggest are kubernetes already has a built-in way to manage this data, secrets, and when moving to CRDs we don't want to create a resource for extracted credentials.
+
+### Problem Description
+The problem is that we should not manage data that is of a sensitive nature if we do not have to. This proposal is limited in scope and only interested in how we save the extracted credentials. It is worth noting that we should eventually be better about how we transmit this data to APBs. 
+
+In the secret, we will save the data in the following format.
+```yaml
+data:
+  DB_PASSWORD: <GOB_ENCODED_VALUE>
+  DB_USERNAME: <GOB_ENCODED_VALUE>
+  ....
+apiVersion: v1
+kind: Secret
+metadata:
```

Comment:
It may be a good to store the ClusterServiceClass externalName (i.e the human readable value) and the action in a label on the secret. The purpose of this would be to assist debugging after the fact. It is likely that over a long period of time or during development, there may be many secrets created. If I am trying to find the secrets created by a particular ServiceClass it would be nice to be able to use the label selector 
Something like:
```
 metadata:
  labels:
    clusterServiceClassName/ apbName : "dh-service-apb"
    action:"provision/bind"

```
Should allow me to do
```
oc get secrets -l apbName=dh-myservice-apb -l action=bind
```

Reply:
This sounds good.

I will add the ability to pass labels to the `runtime` package functions and to the `clients` package functions.